crates/hir_ty/src/diagnostics/expr.rs

```diff
@@ -150,20 +163,58 @@ impl<'a, 'b> ExprValidator<'a, 'b> {
         }
     }
 
-    fn validate_call(&mut self, db: &dyn HirDatabase, call_id: ExprId, expr: &Expr) -> Option<()> {
+    fn check_for_filter_map_next(&mut self, db: &dyn HirDatabase) {
+        let body = db.body(self.owner.into());
+        let mut prev = None;
+
+        for (id, expr) in body.exprs.iter() {
+            if let Expr::MethodCall { receiver, method_name, args, .. } = expr {
+                let method_name_hack_do_not_merge = format!("{}", method_name);
```

Comment:
What's the proper way of doing this?

Reply:
Mayebe `.to_string()`, but I'm not sure.